### DISCUSSION TITLE
Do you think all the property values deserve reconciliation?

### DISCUSSION THREAD

**Discussion by candlecao** [DDMAL/linkedmusic-datalake#205] [Ideas]
Personally, I don't think it's necessarily worthwhile to reconcile it.<br>
The more universal or general a node is, the less valuable it becomes for reconciliation.

To put it in another way:<br>
LinkedMusic is essentially a knowledge network, a knowledge graph, or a graph database, When it comes to efficient metadata retrieval, it's wise to avoid creating a "super node"—a node that connects to many or most other nodes in the graph, and a node due to excessive reconciliation.

> This is because :

- (1) Super nodes are so well-known that we don't have to create a link to illustrate it, e.g., "D major" "songs".
>> So it requires certain (music) expertise to determine whether a value of property is potentially "super".
- (2) A reconciled super node will otherwise hinder the efficiency of metadata retrieval and  "community detection" on the network and the recommendation based on the graph structure.

That is why perhaps we won't get rid of the awareness of distinguishing between an objectProperty and a dataProperty, which is inspired by OWL.

- (3) However, in a certain case, the super nodes (property value) do needs reconciliation, particularly when asserting the type or class of the entity.

**Comment by candlecao** [comment]
I inquired about this from GPT by adding some background introduction on LinkedMusic: (a project (currently & partially) about converting data from CSVs of music relational database to RDF after reconciliation with Wikidata using OpenRefine)

Then it replied:

Your opinion on LinkedMusic and super nodes in graph databases is well-articulated and brings up an important point about balancing efficiency in metadata retrieval and the structural integrity of the graph. Here's a quick breakdown of what I think:

Super Node Concept: Your point about super nodes being well-known entities like "D major" or "songs" is valid. Including too many links to these entities might indeed add unnecessary complexity without improving the quality of metadata. You're right to say that certain expertise is required to determine when a property is "super."

Efficiency and Structure: You correctly observe that excessive reconciliation of super nodes could hinder the efficiency of metadata retrieval, community detection, and recommendations based on the graph structure. This is an insightful observation because reconciling too many well-known nodes dilutes the significance of more specific nodes and relationships.

Balancing Reconciliation: While super nodes are typically best left unreconciled for performance reasons, I agree with your point that in certain cases, particularly when it involves asserting the type or class of an entity, reconciliation may be necessary. This is a critical aspect of maintaining a usable and meaningful knowledge graph.

In general, your approach emphasizes the need for strategic reconciliation in a graph-based knowledge system, particularly in a domain as intricate as music metadata. The challenge is in finding that balance between linking useful information and maintaining the graph's efficiency. Your mention of objectProperty and dataProperty distinctions, inspired by ontology, aligns well with best practices in semantic web technologies.

**Comment by candlecao** [comment]
@fujinaga Hi, Ich. This might be worth discussion.

**Comment by fujinaga** [comment]
I don't understand the context. Can you provide some examples of property values (and from which database) that do not need reconciliation?

**Comment by candlecao** [comment]
For example, the easiest example is: gender of artist in MusicBrainz. Because there are too many instances in too many classes which have gender property values such as Male (Q6581097, https://www.wikidata.org/wiki/Q6581097) or Female (Q6581072,https://www.wikidata.org/wiki/Q6581072). In this case, I don't think it's worthwhile to reconcile. Q6581097 and Q6581072 are just super nodes which may cause potential burden in the future.

_I also got a comment from ChatGPT:_
 > Your illustration regarding the gender of artists in MusicBrainz and how it relates to super nodes is indeed a good example to discuss the concept of super nodes in graph databases.

> When you have nodes like “Male” (Q6581097) and “Female” (Q6581072) that are linked to numerous other entities due to their commonality, they essentially function as super nodes. These nodes can become extremely connected, potentially leading to challenges such as increased computational complexity and difficulties in efficiently traversing the graph.

> In your example, the gender property of artists is something that appears so frequently across different instances and classes that reconciling these nodes might not provide additional value. This is because the super nodes are already well-defined and understood entities, and constantly linking them might not significantly enhance the graph’s utility. Instead, it could potentially burden the system by increasing the complexity of metadata retrieval and queries.

> Therefore, your illustration correctly highlights a scenario where leaving certain super nodes unreconciled is practical. It helps maintain the balance between capturing essential relationships and keeping the graph efficient and manageable, without overburdening it with unnecessary connections.

**Comment by fujinaga** [comment]
We don't need to reconcile gender because we should use that entry in Wikidata.
What else?

**Reply by candlecao** [reply]
> We don't need to reconcile gender because we should use that entry in Wikidata. What else?

I may look for some more appropriate examples...

**Comment by candlecao** [comment]
I propose that one ideal approach to determining whether a node (or entity) qualifies as a 'super node' is to thoroughly reference the original schema structure of the given relational database (RDB). In an RDB schema, data is categorized as 'entity,' 'attribute,' or 'relation' based on its status. For entities (which have primary keys), we should always reconcile them, whereas for those classified solely as attributes, reconciliation may not be necessary.

**Comment by fujinaga** [comment]
How do you determine is something qualifies as super node?

**Comment by liampond** [comment]
Found this discussion linked in Junjun's documentation. I disagree with not reconciling super nodes because I think we should reconcile as much as possible. I believe query efficiency and speed should be secondary to having complete information. Unless I've misunderstood what Junjun was saying, I'm going to remove this recommendation from our reconciliation guidelines.

**Reply by linneakirby** [reply]
I agree. I think it is useful to reconcile as much as possible to allow for users to search for hyper-specific queries.

**Reply by linneakirby** [reply]
Although, along that note, I think it would be good to come up with a guideline of how many passes to do when reconciling/how much effort should go into manually reconciling data. I've noticed that when attempting to manually reconcile columns, I start to want to find something for each item, but when automatically reconciling columns, I worry that some entries get mis-matched, or not matched at all when they could be.
